elastic_monte_carlo_estimate_of_pi.ipynb

```diff
@@ -0,0 +1,358 @@
+   "metadata": {},
+   "source": [
+    "# Monte-Carlo Estimate of $\\pi$\n",
+    "\n",
+    "We want to estimate the number $\\pi$ using a [Monte-Carlo method](https://en.wikipedia.org/wiki/Pi#Monte_Carlo_methods) exploiting that the area of a quarter circle of unit radius is $\\pi/4$ and that hence the probability of any randomly chosen point in a unit square to lie in a unit circle centerd at a corner of the unit square is $\\pi/4$ as well.  So for N randomly chosen pairs $(x, y)$ with $x\\in[0, 1)$ and $y\\in[0, 1)$, we count the number $N_{circ}$ of pairs that also satisfy $(x^2 + y^2) < 1$ and estimage $\\pi \\approx 4 \\cdot N_{circ} / N$.\n",
+    "\n",
+    "[<img src=\"https://upload.wikimedia.org/wikipedia/commons/8/84/Pi_30K.gif\" \n",
+    "     width=\"50%\" \n",
+    "     align=top\n",
+    "     alt=\"PI monte-carlo estimate\">](https://en.wikipedia.org/wiki/Pi#Monte_Carlo_methods)\n",
+    "\n",
+    "## Why Adaptive?\n",
+    "\n",
+    "Using [Dask's adaptivity](http://docs.dask.org/en/latest/setup/adaptive.html), we'll show that it is possible to scale the available resources to meet almost any desired wall times irrespective of the actual work load.  This is important, because it allows for focussing on the situation of the human running the analysis:\n",
+    "\n",
+    "> _\"Am I in an explorative and creative phase of my work where it is important that I can see the next plot within seconds?\"_\n",
+    "\n",
+    "> _\"Am I running routine analyses that can wait until tomorrow or next week?\"_\n",
+    "\n",
+    "## Actual timings\n",
+    "\n",
+    "Aming for a duration of 20 seconds per calculation, this is what we actually get:\n",
+    "\n",
+    "- $\\pi$ from 50.0 GB of random data in 21.82 s with 13 workers\n",
+    "- $\\pi$ from 100.0 GB of random data in 16.73 s with 25 workers\n",
+    "- $\\pi$ from 200.0 GB of random data in 15.79 s with 53 workers\n",
+    "- $\\pi$ from 400.0 GB of random data in 16.94 s with 122 workers\n",
+    "- $\\pi$ from 800.0 GB of random data in 21.84 s with 241 workers\n",
+    "- $\\pi$ from 1600.0 GB of random data in 26.89 s with 400 workers\n",
+    "- $\\pi$ from 3200.0 GB of random data in 45.50 s with 392 workers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Tuning adaptivity\n",
+    "\n",
+    "The following tunes a Dask cluster to use anywhere between 1 and 400 workers and to scale its size so that any computation is finished within 20 seconds.  On Pangeo, time scales for starting / stopping workers are of the order of a few seconds, so we set a startup cost to 5 seconds (instead of the default value of 1 second) and increase possible scale-down times by setting the relevant interval to 2 seconds and the number of times a worker needs to be considered expendable before it is actually killed to `10`.  We also reduce the default factor that is applied to adapt the cluster to a more modest `1.2`.\n",
```

Comment:
>time scales for starting / stopping workers are of the order of a few seconds

Not sure this is true for Cloud deployments, is it?

Why only 1.2 scale factor? I'm not sure this is used in most cases.